api/src/workers/sale-offer-handler.ts

```diff
@@ -1,30 +1,48 @@
+import * as fcl from "@onflow/fcl";
+
 import { BlockCursorService } from "../services/block-cursor";
 import { FlowService } from "../services/flow";
-import { EventDetails, BaseEventHandler } from "./base-event-handler";
 import { MarketService } from "../services/market";
 
-interface SaleOfferCreated {
-  itemID: number;
-  price: number;
-}
+import { EventDetails, BaseEventHandler } from "./base-event-handler";
 
 class SaleOfferHandler extends BaseEventHandler {
+  private eventCollectionInsertedSaleOffer;
+  private eventCollectionRemovedSaleOffer;
   constructor(
-    blockCursorService: BlockCursorService,
-    flowService: FlowService,
     private readonly marketService: MarketService,
-    eventName: string
+    blockCursorService: BlockCursorService,
+    flowService: FlowService
   ) {
-    super(blockCursorService, flowService, eventName);
+    super(blockCursorService, flowService, []);
+
+    this.eventCollectionInsertedSaleOffer = `A.${fcl.sansPrefix(
+      marketService.marketAddress
+    )}.KittyItemsMarket.CollectionInsertedSaleOffer`;
+
+    this.eventCollectionRemovedSaleOffer = `A.${fcl.sansPrefix(
+      marketService.marketAddress
+    )}.KittyItemsMarket.CollectionRemovedSaleOffer`;
+
+    this.eventNames = [
+      this.eventCollectionInsertedSaleOffer,
+      this.eventCollectionRemovedSaleOffer,
+    ];
```

Comment:
I think you can do this as long as `this` isn't accessed/mutated before the call to `super`.

```suggestion
    const eventCollectionInsertedSaleOffer = `A.${fcl.sansPrefix(
      marketService.marketAddress
    )}.KittyItemsMarket.CollectionInsertedSaleOffer`;
    
    const eventCollectionRemovedSaleOffer = `A.${fcl.sansPrefix(
      marketService.marketAddress
    )}.KittyItemsMarket.CollectionRemovedSaleOffer`;
    
    const eventNames = [
      eventCollectionInsertedSaleOffer,
      eventCollectionRemovedSaleOffer,
    ];
    
    super(blockCursorService, flowService, eventNames);

    this.eventCollectionInsertedSaleOffer = eventCollectionInsertedSaleOffer;
    this.eventCollectionRemovedSaleOffer = eventCollectionRemovedSaleOffer;
```

Reply:
No, does not compile :(